src/frontend/packages/semantic-data-provider/dist/index.d.ts

```diff
@@ -73,6 +73,8 @@ export type SemanticDataProvider = DataProvider & {
   getDataModels: () => Promise<Record<string, DataModel>>;
   getDataServers: () => Promise<DataServersConfig>;
   fetch: FetchFn;
+  refreshConfig: () => Promise<Configuration>;
+  uploadFile: (rawFile: File, config: Configuration) => Promise<string | null>;
```

Comment:
Looking at the other methods, it seems you shouldn't include the `config`, since it's already handled by the `waitForConfig` function.

Reply:
Sorry, that was my fault :D

Reply:
That fault was a team effort 😅 
That means we're already acting as team which is very promising ! 🤝